perception/autoware_tensorrt_bevdet/src/postprocess.cu

```diff
@@ -0,0 +1,212 @@
+  const int _output_h, const int _output_w, const float _x_step, const float _y_step,
+  const float _x_start, const float _y_start, const std::vector<int> & _class_num_pre_task,
+  const std::vector<float> & _nms_rescale_factor)
+: class_num(_class_num),
+  score_thresh(_score_thresh),
+  nms_thresh(_nms_thresh),
+  nms_pre_maxnum(_nms_pre_maxnum),
+  nms_post_maxnum(_nms_post_maxnum),
+  down_sample(_down_sample),
+  output_h(_output_h),
+  output_w(_output_w),
+  x_step(_x_step),
+  y_step(_y_step),
+  x_start(_x_start),
+  y_start(_y_start),
+  map_size(output_h * output_w),
+  class_num_pre_task(_class_num_pre_task),
+  nms_rescale_factor(_nms_rescale_factor),
+  task_num(_class_num_pre_task.size())
+{
+  CHECK_CUDA(cudaMalloc((void **)&boxes_dev, kBoxBlockSize * map_size * sizeof(float)));
+  CHECK_CUDA(cudaMalloc((void **)&score_dev, map_size * sizeof(float)));
+  CHECK_CUDA(cudaMalloc((void **)&cls_dev, map_size * sizeof(int)));
+  CHECK_CUDA(cudaMalloc((void **)&sorted_indices_dev, map_size * sizeof(int)));
+  CHECK_CUDA(cudaMalloc((void **)&valid_box_num, sizeof(int)));
+  CHECK_CUDA(cudaMalloc((void **)&nms_rescale_factor_dev, class_num * sizeof(float)));
+
+  CHECK_CUDA(cudaMallocHost((void **)&boxes_host, kBoxBlockSize * map_size * sizeof(float)));
+  CHECK_CUDA(cudaMallocHost((void **)&score_host, nms_pre_maxnum * sizeof(float)));
+  CHECK_CUDA(cudaMallocHost((void **)&cls_host, map_size * sizeof(float)));
+  CHECK_CUDA(cudaMallocHost((void **)&sorted_indices_host, nms_pre_maxnum * sizeof(int)));
+  CHECK_CUDA(cudaMallocHost((void **)&keep_data_host, nms_pre_maxnum * sizeof(std::int64_t)));
+
+  CHECK_CUDA(cudaMemcpy(
+    nms_rescale_factor_dev, nms_rescale_factor.data(), class_num * sizeof(float),
+    cudaMemcpyHostToDevice));
+
+  iou3d_nms.reset(new Iou3dNmsCuda(output_h, output_w, nms_thresh));
+
+  std::ostringstream oss;
+  for (auto i = 0; i < nms_rescale_factor.size(); i++) {
+    oss << std::fixed << std::setprecision(2) << nms_rescale_factor[i] << ' ';
+  }
+  RCLCPP_INFO(rclcpp::get_logger("PostprocessGPU"), "%s", oss.str().c_str());
+}
+PostprocessGPU::~PostprocessGPU()
+{
+  CHECK_CUDA(cudaFree(boxes_dev));
+  CHECK_CUDA(cudaFree(score_dev));
+  CHECK_CUDA(cudaFree(cls_dev));
+  CHECK_CUDA(cudaFree(sorted_indices_dev));
+  CHECK_CUDA(cudaFree(valid_box_num));
+  CHECK_CUDA(cudaFree(nms_rescale_factor_dev));
+
+  CHECK_CUDA(cudaFreeHost(boxes_host));
+  CHECK_CUDA(cudaFreeHost(score_host));
+  CHECK_CUDA(cudaFreeHost(cls_host));
+  CHECK_CUDA(cudaFreeHost(sorted_indices_host));
+  CHECK_CUDA(cudaFreeHost(keep_data_host));
+}
+
+void PostprocessGPU::DoPostprocess(void ** const bev_buffer, std::vector<Box> & out_detections)
+{
+  // bev_buffer : BEV_feat, reg_0, hei_0, dim_0, rot_0, vel_0, heatmap_0, reg_1 ...
+  const int task_num = class_num_pre_task.size();
+  int cur_start_label = 0;
+  for (int i = 0; i < task_num; i++) {
+    float * reg = (float *)bev_buffer[i * 6 + 0];      // 2 x 128 x 128
```

Comment:
The comments indicating the dimensions (e.g., `// 2 x 128 x 128`) should be replaced with more descriptive variable names or constants that clearly define these dimensions.  Hardcoding these values makes the code less flexible and harder to maintain.